### DISCUSSION TITLE
Equivalent of MySQL.Data's static methods

### DISCUSSION THREAD

**Discussion by Arthri** [mysql-net/MySqlConnector#1019] [Q&A] [answered]
Hi! I'm migrating from MySQL.Data/MySQL Connector.NET and I was wondering if there's an equivalent class to MySQL.Data's MySqlHelper. The class basically provides static methods for querying, a connection string can be passed in, and in that case it will manage connections pools automatically.

https://dev.mysql.com/doc/connector-net/en/connector-net-connections-pooling.html

If I'm right, I should just create a new `MySqlConnection` every time I need to query, and `OpenAsync()` manages the pools automatically right?

**Comment by bgrainger** [comment, marked as answer]
Correct; all `MySqlConnection` objects created with the same connection string are automatically pooled (unless it contains `Pooling=false`).

**Reply by Arthri** [reply]
Thank you for the answer ♥

**Comment by lijinv** [comment]
I'm also in the process of migrating from MySql.Data's MySqlHelper to MySqlConnector. I'm watching the Connection Open counts and even though Pooling is set to True, the # of Opens = the # of SP calls being made. The code is calling OpenAsync() every time but I would assume actual Opens are lower because of pooling. Am I missing anything?

**Reply by lijinv** [reply]
Any thoughts @bgrainger? 

This is the code block of it helps:

await using var connection = new MySqlConnection(connectionString);
            await connection.OpenAsync(connectionToken).ConfigureAwait(false);
            await using var command = new MySqlCommand(commandText, connection);
            foreach (var p in parameters)
                command.Parameters.Add(new MySqlParameter(p.ParameterName, p.Value));
            await command.ExecuteNonQueryAsync(commandToken).ConfigureAwait(false);

**Reply by bgrainger** [reply]
How are you "watching the Connection Open counts"?

"I would assume actual Opens are lower because of pooling." Yes, this should be the case. `MySqlConnection.Open` retrieves a connection from the pool, checks its health, and then returns it. It doesn't usually initiate a new connection (i.e., a new TCP socket) to the server.

**Reply by lijinv** [reply]
Connection Opens are tracked in our Observability platform (Dynatrace)

**Reply by bgrainger** [reply]
As per https://github.com/mysql-net/MySqlConnector/issues/1036:

> * `Open` - establishing a connection to a database server
>   * The actual implementation may simply be retrieving a connection from a pool; this is still recorded as an `Activity`.

This is most likely what you're observing.
